src/main/java/bowling/controller/BowlingGameController.java

```diff
@@ -31,18 +30,26 @@ public BowlingGameResponse pitchBall(BowlingGameRequest request) {
     }
 
     private BowlingGameResponse assembleGameResponse(Participant participant, Frames frames) {
-        return new BowlingGameResponse(participant.getName(), frames.nextTurnNumber(), frames.isFinished(), assemblePinDownResults(frames));
+        return new BowlingGameResponse(participant.getName(), frames.nextTurnNumber(), assembleFrameInfos(frames), frames.isFinished());
     }
 
-    private List<String> assemblePinDownResults(Frames frames) {
-        List<String> pinDownResults = new ArrayList<>();
+    private List<FrameInfo> assembleFrameInfos(Frames frames) {
+        List<FrameInfo> frameInfos = new ArrayList<>();
         for (Frame frame : frames) {
-            pinDownResults.add(assemblePinDownResult(frame.getScoreBoards()));
+            frameInfos.add(assembleFrameInfo(frame));
```

Comment:
```suggestion
            FrameInfo frameInfo = frame.assembleFrameInfo();
            frameInfos.add(frameInfo);
```
```java
//class Frame
public FrameInfo assembleFrameInfo() {
    return new FrameInfo(assemblePinDownResults(pitches()), score(), isFinalFrame(), isSpare(), isScoreDecided());
}

private List<Integer> assemblePinDownResults(Pitches pitches) {
    List<Integer> pinDownResults = new ArrayList<>();
    for (Pitch pitch : pitches) {
        pinDownResults.add(pitch.value());
    }
    return pinDownResults;
}
```
Frame이 FrameInfo를 만들어서 반환하는게 더 적절한 책임으로 보이는데 어떤가요? 🙂

Reply:
비슷한 맥락으로 assembleGameResponse() 메서드를 Frames가 처리하는 건 어떤가요? 전체적으로 반영하면 아래와 같은 코드가 되겠습니다.

```java
//class BowlingGameController
public BowlingGameResponse startGame(BowlingGameRequest request) {
    Participant participant = new Participant(request.getParticipantName());
    service.startGame(participant);
    Frames frames = service.findFrames(participant);
    return frames.assembleGameResponse(participant);
}

public BowlingGameResponse pitchBall(BowlingGameRequest request) {
    Participant participant = new Participant(request.getParticipantName());
    service.pitchBall(participant, request.getPitchCount());
    Frames frames = service.findFrames(participant);
    return frames.assembleGameResponse(participant);
}
```
```java
//class Frames
public BowlingGameResponse assembleGameResponse(Participant participant) {
    return new BowlingGameResponse(participant.getName(), nextTurnNumber(), assembleFrameInfos(), isFinished());
}

private List<FrameInfo> assembleFrameInfos() {
    List<FrameInfo> frameInfos = new ArrayList<>();
    for (Frame frame : frames) {
        FrameInfo frameInfo = frame.assembleFrameInfo();
        frameInfos.add(frameInfo);
    }
    return frameInfos;
}
```

Reply:
네 시윤님 코멘트 감사합니다~! 제가 아직 DDD 아키텍처에 대한 공부가 짧고, 실무에서 이런 구조에 대한 고민을 많이 안해봐서 뭔가 도메인 영역에서 dto를 사용한다는 것에 대해 거부감이 좀 있었던 것 같습니다!

가이드주신 부분을 읽어보니, 연관된 도메인 객체에서 추출될 수 있는 dto는 도메인 객체가 직접 만들어서 리턴해주는게 더 좋겠다는 생각이 드네요! 다시보니 dto의 구성도 Frame 정보와 Frames 에서 추출할 수 있는 구성으로 이루어져 있어서 시윤님 가이드대로 구현해보면 컨트롤러 소스가 훨씬 가벼워 질 것 같습니다. 의견주신대로 한 번 수정해 보겠습니다! 감사합니다!